htdocs/comm/propal/card.php

```diff
@@ -1163,38 +1163,9 @@
 						$object->generateDocument($object->model_pdf, $outputlangs, $hidedetails, $hidedesc, $hideref);
 					}
 
-					unset($_POST['prod_entry_mode']);
-
-					unset($_POST['qty']);
-					unset($_POST['type']);
-					unset($_POST['remise_percent']);
-					unset($_POST['price_ht']);
-					unset($_POST['multicurrency_price_ht']);
-					unset($_POST['price_ttc']);
-					unset($_POST['tva_tx']);
-					unset($_POST['product_ref']);
-					unset($_POST['product_label']);
-					unset($_POST['product_desc']);
-					unset($_POST['fournprice']);
-					unset($_POST['buying_price']);
-					unset($_POST['np_marginRate']);
-					unset($_POST['np_markRate']);
-					unset($_POST['dp_desc']);
-					unset($_POST['idprod']);
-					unset($_POST['units']);
-
-					unset($_POST['date_starthour']);
-					unset($_POST['date_startmin']);
-					unset($_POST['date_startsec']);
-					unset($_POST['date_startday']);
-					unset($_POST['date_startmonth']);
-					unset($_POST['date_startyear']);
-					unset($_POST['date_endhour']);
-					unset($_POST['date_endmin']);
-					unset($_POST['date_endsec']);
-					unset($_POST['date_endday']);
-					unset($_POST['date_endmonth']);
-					unset($_POST['date_endyear']);
+					$urltogo = preg_replace('/--IDFORBACKTOPAGE--/', $object->id, $urltogo) . '#addline';
```

Comment:
I am not a big fan of making redirect when it is not absolutely necessary. It increase the "flash" effect on slow servers, making a bad experience for end user.
Why not instead, introducing a function unsetPostAfterInsertLine() ?